fixit/common/config.py

```diff
@@ -43,6 +43,10 @@
     + r": (?P<codes>([-_a-zA-Z0-9]+,\s*)*[-_a-zA-Z0-9]+)"
     + r"(:( (?P<reason>.+)?)?)?"
 )
+HAS_LINT_IGNORE_REGEXP: Pattern[bytes] = re.compile(rb"# lint-(ignore|fixme):")
+HAS_LINT_IGNORE_OR_NOQA_REGEXP: Pattern[bytes] = re.compile(
+    rb"# (lint-(ignore|fixme):|noqa)"
+)
```

Comment:
This doesn't seem fully correct. E.g. FLAKE8_NOQA_FILE (`# flake8[:=]\s*noqa`) is not covered.
Can you refactor patterns (`FLAKE8_NOQA_FILE`, `NOQA_FILE_RULE`) by extracting the string argument of `re.compile` and reusing them in `HAS_LINT_IGNORE_OR_NOQA_REGEXP` pattern?
That way we can make sure the new pattern covers all patterns used in `IgnoreInfo.compute()` better.

Reply:
Thanks Jimmy for catching that case. I added a test and fixed the regex.

Regarding how to refactor this in order to ensure all patterns are covered, I think that extracting a string to be reused could complicated the code quite a bit.

Maybe a better approach would be to just construct `HAS_LINT_IGNORE_OR_NOQA_REGEXP` based on the other patterns. I avoided this originally, as those regexes are complicated and may be slow in some cases.

I will add a suggestion to the code so you can see what I mean.